mlir/lib/Dialect/Quant/IR/QuantTypes.cpp

```diff
@@ -660,19 +662,18 @@ LogicalResult QuantileQuantizedPerAxisType::verifyInvariants(
   unsigned typeWidth{};
   if (storageType.isa<IntegerType>()) {
     typeWidth = llvm::dyn_cast<IntegerType>(storageType).getWidth();
-  } else if (storageType.isa<Float8E5M2Type>() ||
-             storageType.isa<Float8E4M3FNType>()) {
-    // Both Float8E5M2Type and Float8E4M3FNType derive from FloatType.
+  } else if (mlir::isa<Float8E5M2Type, Float8E4M3FNType, Float4E2M1FNType>(storageType)) {
```

Comment:
nitpick:
```suggestion
  } else if (mlir::isa<FloatType>(storageType)) {
```

this seems sufficient here.

Reply:
It should check only for supported types. It will be simplified once the interface based approach will be introduced with Roman's PR.